### ISSUE TITLE
treeanc.py throws error on augur build of american-only Zika genomes

### ISSUE THREAD

**Issue by alliblk** [neherlab/treetime#27]
When I try to do a `nextstrain/augur` build with a fasta that includes Zika genomes only sampled from the Americas, the build breaks with this error:

  `File "../treetime_augur/treetime/treeanc.py", line 297, in make_reduced_alignment`
   ` n.cseq = self.reduced_alignment[seq_count]`
`IndexError: index 5 is out of bounds for axis 0 with size 5`

I think this might be an error in how the tree traversal is occurring when assigning the compressed sequences, but that's just a guess.

The build and the ancestral state reconstruction works fine however if I include sequences from French Polynesia in addition to the Americas only sequences. Do you have an idea of what might be going on here?

I've attached both the Americas + French Polynesia file (build that works) and the Americas only file (build that doesn't work) here.

[augur-americas-andfp.txt](https://github.com/neherlab/treetime/files/1316199/augur-americas-andfp.txt)
[augur-americas-only.txt](https://github.com/neherlab/treetime/files/1316200/augur-americas-only.txt)

**Comment by alliblk** [comment]
@trvrb thinks that this issue might be cropping up when TreeTime tries to assign ancestral states to a column with all missing data. I think this may in fact be the case as an augur build with only Americas sequences and the reference sequence (which is a complete genome) works.

**Comment by trvrb** [comment]
@rneher ---

Here's a very fast way to reproduce the bug. Save and rename the attached `zika.json.txt` to `augur/zika/prepared/zika.json` (GitHub doesn't like JSON endings). Then run `zika.process.py` from `augur/zika/`.

[zika.json.txt](https://github.com/neherlab/treetime/files/1326933/zika.json.txt)

I get the following error:

```
Traceback (most recent call last):
  File "zika.process.py", line 63, in <module>
    runner.run_geo_inference()
  File "../../base/process.py", line 702, in run_geo_inference
    self.tree.geo_inference(geo_attr, **kwargs)
  File "../../base/tree.py", line 321, in geo_inference
    self.tt.make_reduced_alignment()
  File "../../treetime_augur/treetime/treeanc.py", line 297, in make_reduced_alignment
    n.cseq = self.reduced_alignment[seq_count]
IndexError: index 5 is out of bounds for axis 0 with size 5
```

So, geo inference not ancestral state reconstruction. It's possible this is an augur bug rather than a TreeTime bug. But if you could take a look it would be much appreciated.

**Comment by rneher** [comment]
the problem in this example was that some terminal nodes (PF13/251013_18) didn't have a region assigned. I also make treetime check explicitly that all sequences have the same length. 

fixed in commit 514a370314bc0164a03433497fe95f911010373a

**Comment by trvrb** [comment]
Oh. I see now. Thanks so much for fixing this @rneher.
